File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/messages/single-sign-on-message.ts
import {IActionItem} from '../actions/action-item.interface';
import {OpenposMessage} from './message';
import {MessageTypes} from './message-types';

export class SingleSignOnMessage implements OpenposMessage {
    type = MessageTypes.SINGLE_SIGN_ON_REQUEST;
    providerName: string;
    responseAction: IActionItem;
}

File: openpos-flow/src/main/java/org/jumpmind/pos/core/ui/SingleSignOnMessage.java
package org.jumpmind.pos.core.ui;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.jumpmind.pos.core.model.MessageType;
import org.jumpmind.pos.util.model.Message;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SingleSignOnMessage extends Message {
    private String providerName;
    private ActionItem responseAction;
    @Override
    public String getType() {
        return MessageType.SingleSignOnRequest;
    }
}
